src/features/environmentCreate/components/Specification/SpecificationCreate.tsx

```diff
@@ -59,21 +66,27 @@ export const SpecificationCreate = ({ onCreateEnvironment }: any) => {
       dispatch(
         editorCodeUpdated({
           channels: editorContent.channels,
-          dependencies: editorContent.dependencies
+          dependencies: editorContent.dependencies,
+          variables: editorContent.variables
         })
       );
     } else {
-      setEditorContent({ dependencies: requestedPackages, channels });
+      setEditorContent({
+        dependencies: requestedPackages,
+        variables,
+        channels
+      });
     }
 
     setShow(value);
   };
 
-  const formatCode = (channels: string[], dependencies: string[]) => {
-    if (channels.length === 0 && dependencies.length === 0) {
-      return "channels:\n  -\ndependencies:\n  -";
-    }
-    return stringify({ channels, dependencies: dependencies });
+  const formatCode = (
+    channels: string[],
+    dependencies: string[],
+    variables: Record<string, string>
+  ) => {
+    return stringify({ channels, dependencies, variables });
```

Comment:
I've removed custom pretty-printing code that we had before. I don't see much value in it because:
- it will be ignored if either value is not empty since it uses `&&`
- with the newly-added `variables` field, the number of possible empty/non-empty variations is even bigger
- what `stringify` returns is valid YAML and I think it's readable.

You can see the old version in the diff.

The new version looks like this:
```yaml
channels: []
dependencies: []
variables: {}
```

Reply:
I probably should replace this with `stringify` wherever this is called because we're doing this in `src/features/environmentDetails/components/Specification/SpecificationEdit.tsx` anyway.